caddyhttp/httpserver/plugin.go

```diff
@@ -51,7 +51,8 @@ var contexts []*httpContext
 
 func newContext() caddy.Context {
 	context := &httpContext{keysToSiteConfigs: make(map[string]*SiteConfig)}
-	contexts = append(contexts, context)
+	// put the new context at start to allow setup of directives on new instance
+	contexts = append([]*httpContext{context}, contexts...)
```

Comment:
I wonder: should the context have some attachment to or knowledge of the Instance so that the order doesn't matter? I am trying to see if there is a way to do it without having to pass it into httpserver.GetConfig() which is obtained from the Controller, along with the Key.... yuck...